atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/CachingTransaction.java

```diff
@@ -137,6 +140,37 @@ public Transaction delegate() {
         return cacheHit;
     }
 
+    @Override
+    public ListenableFuture<Map<Cell, byte[]>> getAsync(TableReference tableRef, Set<Cell> cells) {
+        if (cells.isEmpty()) {
```

Comment:
Feels like we should probably refactor some shared code here with #get instead of copying.

Reply:
PR #4304, resolves this issue.